cpp/include/cudf/detail/gather.cuh

```diff
@@ -255,13 +256,106 @@ struct column_gatherer_impl<dictionary32, MapItType> {
   }
 };
 
-/*
+/**
+ * @brief Column gather specialization for list_view column type.
+ *
+ * @tparam MapItRoot Iterator type to access the incoming root column.
+ *
+ * This functor is invoked only on the root column of a hierarchy of list
+ * columns. Recursion is handled internally.
+ */
+template <typename MapItRoot>
+struct column_gatherer_impl<list_view, MapItRoot> {
+  /**
+   * @brief Gather a list column from a hierarchy of list columns. This is the start
+   * of the recursion - we will only ever get in here once.
+   *
+   * This function is looks similar to gather_list_nested() but the difference is
```

Comment:
"This function is looks similar to" should be "This functions looks similar to"...